src/app/destiny1/activities/Activities.tsx

```diff
@@ -79,7 +79,7 @@ export default function Activities({ account }: Props) {
       tier.activityData.recommendedLight === 390
         ? '390'
         : tier.tierDisplayName
-        ? t(`Activities.${tier.tierDisplayName}`, {
+        ? t(`Activities.${tier.tierDisplayName}` as I18nKey, {
```

Comment:
I'm not sure I know a better way to handle these. Perhaps if things like `tierDisplayName` were modeled as string union types?

Reply:
The TS type checker can certainly handle that functionality by distributing the interpolation across the string union.

> extra typechecking time

I was thinking "how bad can it be?" but when playing around with it this added an extra 6s of latency to some IDE hover type hints, which is pretty extreme. I wonder if this is expected or just some unoptimized pathological case in the TS type checker...

Reply:
Looking at https://github.com/i18next/i18next/issues, parts of this feature seem to be really heavy on the TS type checker and cause a variety of errors (OOMs, type instantiation recursion limits, ...) in some projects.

Reply:
Yeah, they've been working on it for several releases. This is the first time it's actually worked.